### PR TITLE
gc error process

### PR DESCRIPTION
**Pull request by huiwq1990**
Signed-off-by: huiwq1990 <huiwq1990@163.com>

/kind enhancement
> /kind good-first-issue
> /kind feature
> /kind question
> /kind design
> /sig ai
> /sig iot
> /sig network
> /sig storage

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: pkg/yurthub/gc/gc.go [hunk 1]
```diff
@@ -99,7 +100,7 @@ func (m *GCManager) gcPodsWhenRestart() error {
 	cfg := m.restConfigManager.GetRestConfig(true)
 	if cfg == nil {
 		klog.Errorf("could not get rest config, so skip gc pods when restart")
-		return err
+		return errors.New("could not get rest config, so skip gc pods when restart")
```
Comment (Congrool) [inline pkg/yurthub/gc/gc.go:103]: It seems that we do not care about error returned from `gcPodsWhenRestart`. Could we modify this function as
```go
func (m *GCManager) gcPodsWhenRestart()
```

Comment (rambohe-ch) [reply pkg/yurthub/gc/gc.go:103]: +1

Comment (rambohe-ch) [reply pkg/yurthub/gc/gc.go:103]: > It seems that we do not care about error returned from `gcPodsWhenRestart`. Could we modify this function as
> 
> ```go
> func (m *GCManager) gcPodsWhenRestart()
> ```

agree +1

File: pkg/yurthub/gc/gc.go [hunk 2]
```diff
@@ -89,29 +89,29 @@ func (m *GCManager) Run() {
 	}, m.eventsGCFrequency, 2, true, m.stopCh)
 }
 
-func (m *GCManager) gcPodsWhenRestart() error {
+func (m *GCManager) gcPodsWhenRestart() {
 	localPodKeys, err := m.store.ListKeys("kubelet/pods")
 	if err != nil || len(localPodKeys) == 0 {
-		return nil
```
Comment (Congrool) [inline pkg/yurthub/gc/gc.go:95]: Could we add a log here, since we do not gc in this path.

**Review by rambohe-ch — APPROVED** [review]

**Pull request merged**